kong/init.lua

```diff
@@ -200,6 +200,11 @@ do
   is_dbless = function(config)
     return config.database == "off"
   end
+
+  is_running_with_db = function(config)
+    return config.role == "control_plane" or
+      (config.role == "traditional" and config.database ~= "off")
+  end
 end
```

Comment:
How is this different from `not is_db_less` (control plane is not dbless, nor is traditional, and in both cases you have database ~= "off")?

Reply:
I overcomplicated it. You are right.
I removed the entire function itself. Please take another look.